dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java

```diff
@@ -727,6 +730,65 @@ public DiscoverResult search(Context context, DiscoverQuery discoveryQuery )
         }
     }
 
+    private class SearchIterator implements Iterator<Item> {
```

Comment:
As this is a private class, we should add JavaDocs to describe it & what it's used for.